pennylane/measurements/measurements.py

```diff
@@ -693,3 +695,40 @@ def map_wires(self, wire_map: dict):
         else:
             new_measurement._wires = Wires([wire_map.get(wire, wire) for wire in self.wires])
         return new_measurement
+
+
+class SampleMeasurement(MeasurementProcess, ABC):
+    """Sample-based measurement process."""
+
+    @abstractmethod
+    def process(
+        self, samples: Sequence[complex], shot_range: Tuple[int] = None, bin_size: int = None
+    ):
+        """Process the given samples.
+
+        Args:
+            samples (Sequence[complex]): computational basis samples generated for all wires
+            shot_range (tuple[int]): 2-tuple of integers specifying the range of samples
+                to use. If not specified, all samples are used.
+            bin_size (int): Divides the shot range into bins of size ``bin_size``, and
+                returns the measurement statistic separately over each bin. If not
+                provided, the entire shot range is treated as a single bin.
+        """
```

Comment:
Good call! Done.

Reply:
It still prevents calls like `super().process()` from the subclass. Anyways, this comment is minor.

Reply:
I see what you mean. If you don't mind, I will add the `NotImplementedError` in PR#2, because then I can test that `super().process_samples` raises an error.